client/src/components/RelatedItems/OutfitItemsList.jsx

```diff
@@ -1,26 +1,72 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 // import styled from 'styled-components';
 import OutfitItemsEntry from 'RelatedItems/OutfitItemsEntry.jsx';
 import Carousel from 'App/Carousel.jsx'
-import { products } from 'tests/testData.js';
+import axios from 'axios';
 
-// need to pass {currentItemId} into the props
-function OutfitItemsList() {
-  // get related items from the API; [array of item id]
-  // get related items detail from the API; [{item 1}, {item 2}..]
-  const OutfitItems = products.slice();
+function OutfitItemsList({ currentItemId }) {
+  const [outfitItems, setOutfitItems] = useState([]);
+  const [currentItem, setCurrentItem] = useState({});
 
-  const OutfitItemsEntries = OutfitItems.map((item) => (
+  useEffect(() => {
+    // function fetchOutfit to get the current product information
+    const fetchOutfit = async (itemId) => {
+      const currentProductResponse = await axios.get(`products/${itemId}`);
+      const currentProduct = currentProductResponse.data;
+      const styles = await axios.get(`/products/${itemId}/styles`);
+      const currentImg = styles.data.results[0].photos[0].url;
+      currentProduct.img = currentImg;
+      setCurrentItem(currentProduct);
+    };
+
+    fetchOutfit(currentItemId);
+  }, [currentItemId]);
+
+  const handleAddOutfitClick = (itemId) => {
+    const outfitItemsCopy = [...outfitItems];
+    // check if current item is already in outfit
+    let currentItemInOutfit = false;
+    outfitItemsCopy.forEach((item) => {
+      if (item.id === itemId) {
+        currentItemInOutfit = true;
+      }
+    });
+    if (!currentItemInOutfit) { outfitItemsCopy.push(currentItem); }
+    setOutfitItems(outfitItemsCopy);
```

Comment:
shorter way:
```
let currentItemInOutfit = outfitItems.any((item) => item.id === itemId);
if (!currentItemInOutfit) { setOutfitItems([...outfitItems, currentItem]); }
```

Reply:
I tried any but it says any is not a function so I tried some instead. but good suggestion!

Reply:
ah yeah I'm always forgetting that it's actually `.some`. good call